### PR TITLE
Mapped shape operator

### PR DESCRIPTION
**Pull request by chxin66**
Signed-off-by: Chen Xin <jack.chen@verisilicon.com>

### CODE REVIEW

File: delegate_main.h [hunk 1]
```diff
@@ -114,20 +122,14 @@ class Delegate {
   void CreateCacheOp(const OpData& op_data);
 
   std::vector<std::shared_ptr<tim::vx::Operation>>& GetOps() { return ops_; }
+  struct OperationDataType GetOp_info() { return op_info_; }
```
Comment (sunshinemyson) [inline delegate_main.h:125]: GetOperationOutput( index : uint32_t) : uint32_t

return index in tflite.

**Review by sunshinemyson — CHANGES_REQUESTED** [review]

File: delegate_main.h (hunk 1 above)
Comment (chxin66) [reply delegate_main.h]: done

File: op_map.cc [hunk 2]
```diff
@@ -2747,6 +2748,35 @@ struct Conv3dMapper : public Conv3dKind<TfLiteConv3DParams> {
...
+                   std::vector<std::shared_ptr<tim::vx::Tensor>>& inputs,
+                   std::vector<std::shared_ptr<tim::vx::Tensor>>& outputs,
+                   const void* params) override {
+    TFLITE_LOG(TFLITE_LOG_INFO, "Creating Shape op");
+    std::vector<uint32_t> shape = inputs[0]->GetShape();
+    tim::vx::TensorSpec shape_spec(tim::vx::DataType::INT32,
+                                   {shape.size()},
+                                   tim::vx::TensorAttribute::CONSTANT);
+    auto shape_tensor = delegate->GetGraph()->CreateTensor(shape_spec, shape.data());
+    delegate->GetTensors()[delegate->GetOperationOutput()[0]] = shape_tensor;
```
Comment (sunshinemyson) [inline op_map.cc:2775]: why not pass index as a parameter? then we don't have to return all the data.

**Review by sunshinemyson — CHANGES_REQUESTED** [review]

File: delegate_main.h [hunk 3]
```diff
@@ -114,6 +114,7 @@ class Delegate {
   void CreateCacheOp(const OpData& op_data);
 
   std::vector<std::shared_ptr<tim::vx::Operation>>& GetOps() { return ops_; }
+  int GetOperationOutput() { return op_info_.outputs[0]; }
```
Comment (sunshinemyson) [inline delegate_main.h:117]: why always return first output tensor? You can add an index parameter for this api then you can get any output you want.

File: op_map.cc [hunk 4]
```diff
@@ -2537,11 +2538,11 @@ struct PackMapper : public OpMapperBase<TfLitePackParams> {
...
-    if (input_tensor.type == kTfLiteInt32 ||
-        (input_tensor.dims->size == 1 && (input_tensor.type == kTfLiteInt8 ||
-                                          input_tensor.type == kTfLiteUInt8))) {
-      return false;
-    }
+    // if (input_tensor.type == kTfLiteInt32 ||
+    //     (input_tensor.dims->size == 1 && (input_tensor.type == kTfLiteInt8 ||
+    //                                       input_tensor.type == kTfLiteUInt8))) {
+    //   return false;
+    // }
```
Comment (sunshinemyson) [inline op_map.cc:2545]: don'e mess up features in single commit. Create a standalone commit for this.

**Review by sunshinemyson — CHANGES_REQUESTED** [review]

**Review by sunshinemyson — APPROVED** [review]

**Pull request merged**